### PR TITLE
Mcd startup cleanup further

### PR DESCRIPTION
**Pull request by cgwalters**
More patches like https://github.com/openshift/machine-config-operator/pull/862

### CODE REVIEW

File: pkg/daemon/update.go [hunk 1]
```diff
@@ -810,19 +810,6 @@ func (dn *Daemon) updateOS(config *mcfgv1.MachineConfig) error {
 	return nil
 }
 
-// RHEL 7.6 logger (util-linux) doesn't have the --journald flag
-func (dn *Daemon) isLoggingToJournalSupported() bool {
-	if dn.OperatingSystem == machineConfigDaemonOSRHEL {
```
Comment (cgwalters) [inline pkg/daemon/update.go]: One thing I wanted to call out...this conditional looked wrong to me, so I inverted it above.

Comment (runcom) [reply pkg/daemon/update.go]: This was actually completely wrong.

Comment (runcom) [reply pkg/daemon/update.go]: Can you create a separate PR for this?

Comment (runcom) [reply pkg/daemon/update.go]: it was meant to compare against RHCOS, not OSRHEL

File: pkg/daemon/daemon.go [hunk 2]
```diff
@@ -208,62 +214,55 @@ func New(
...
+		bootID, err = getBootID()
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to read boot ID")
+		}
+	}
+
+	// RHEL 7.6 logger (util-linux) doesn't have the --journald flag
+	loggerSupportsJournal := true
+	if !mock {
+		if operatingSystem == machineConfigDaemonOSRHEL {
```
Comment (runcom) [inline pkg/daemon/daemon.go]: can we just have like before:

```
if operatingsystem == RHCOS { return true }
else
check --journald
```

Comment (runcom) [reply pkg/daemon/daemon.go]: (where the bug was `s/OSRHEL/RHCOS`)

Comment (cgwalters) [reply pkg/daemon/daemon.go]: The code is now in the constructor so we can't `return true`.

File: pkg/daemon/update.go (hunk 1 above)
Comment (cgwalters) [reply pkg/daemon/update.go]: Done in https://github.com/openshift/machine-config-operator/pull/867

Comment (runcom) [reply pkg/daemon/update.go]: Thanks a lot Colin!

**Comment by cgwalters** [comment]
Rebased :surfing_man:

**Pull request merged**